packages/salesforcedx-vscode-apex-debugger/src/index.ts

```diff
@@ -102,21 +119,54 @@ function registerCommands(): vscode.Disposable {
     'sfdx.debug.exception.breakpoint',
     configureExceptionBreakpoint
   );
+
   const startSessionHandler = vscode.debug.onDidStartDebugSession(session => {
+    areYouThereTimer = areYouStillDebugging(session);
+
     cachedExceptionBreakpoints.forEach(breakpoint => {
       const args: SetExceptionBreakpointsArguments = {
         exceptionInfo: breakpoint
       };
       session.customRequest(EXCEPTION_BREAKPOINT_REQUEST, args);
     });
   });
+  const terminateSessionHandler = vscode.debug.onDidTerminateDebugSession(
+    session => {
+      if (areYouThereTimer) {
+        clearInterval(areYouThereTimer);
+      }
+    }
+  );
   return vscode.Disposable.from(
     customEventHandler,
     exceptionBreakpointCmd,
-    startSessionHandler
+    startSessionHandler,
+    terminateSessionHandler
   );
 }
 
+export function areYouStillDebugging(
+  session: vscode.DebugSession
+): NodeJS.Timer | undefined {
+  return setInterval(async () => {
+    const innerTimer = setTimeout(() => {
+      session.customRequest(IDLE_SESSION_REQUEST);
```

Comment:
@DatGuyJonathan From a usability point this is worrisome. Microsoft argues a warning message must be confirmed by a user. That's why there is no programmatic way to close those. Thus, as long as the message is visible and the user has not made a decision yet we must keep the session alive. We are suggesting a continue is possible by presenting the option. Silently timing out is not good (IMO).